src/Compilers/CSharp/Test/Emit2/Attributes/AttributeTests.cs

```diff
@@ -587,7 +587,228 @@ public static void Main()
 
             var program = (NamedTypeSymbol)comp.GetMember("Program");
             var attributeData = (SourceAttributeData)program.GetAttributes()[0];
-            Assert.Equal(new[] { 0, -1 }, attributeData.ConstructorArgumentsSourceIndices);
+            Assert.True(attributeData.ConstructorArgumentsSourceIndices.IsDefault);
+
+            var attributeSyntax = (AttributeSyntax)attributeData.ApplicationSyntaxReference.GetSyntax();
+            Assert.Equal("a: true", attributeData.GetAttributeArgumentSyntax(parameterIndex: 0, attributeSyntax).ToString());
+            Assert.Equal(@"b: new object[] { ""Hello"", ""World"" }", attributeData.GetAttributeArgumentSyntax(parameterIndex: 1, attributeSyntax).ToString());
+        }
+
+        [Fact]
+        [WorkItem(20741, "https://github.com/dotnet/roslyn/issues/20741")]
+        public void TestComplexOrderedAttributeArguments()
+        {
+            var comp = CreateCompilationWithMscorlib46(@"
+using System;
+
+sealed class MarkAttribute : Attribute
+{
+    public MarkAttribute(bool a, bool b, bool c)
```

Comment:
>bool a, bool b, bool c [](http://example.com/codeflow?start=25&length=22)

Perhaps use `int` rather than `bool` so the three parameters can have distinct values at the use-site.